cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/indexer/IndexerFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.metadata.indexer;
+
+import com.google.common.base.Throwables;
+
+/**
+ * A factory for {@link Indexer}
+ */
+public class IndexerFactory {
```

Comment:
I will change it to just new but my thoughts were to have a factory which can provide indexers to everyone rather than newing different indexers all over the place.